envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoySnapshotFactory.kt

```diff
@@ -64,6 +76,20 @@ internal class EnvoySnapshotFactory(
         )
     }
 
+    private fun toClusterConfiguration(instances: List<ServiceInstances>, serviceName: String): ClusterConfiguration {
+        val allInstances = instances.flatMap {
+            it.instances
+        }
+
+        // Http2 support is on a cluster level so if someone decides to deploy a service in dc1 with envoy and in dc2
+        // without envoy then we can't set http2 because we do not know if the server in dc2 supports it.
+        val allInstancesHaveEnvoyTag = allInstances.isNotEmpty() && allInstances.all {
+            it.tags.contains("envoy")
```

Comment:
I think we should use tag name from property `envoy-control.envoy.snapshot.egress.http2Properties.tagName`

Reply:
You're right, good catch :)